subsys/net/ip/net_if.c

```diff
@@ -4196,6 +4205,65 @@ int net_if_up(struct net_if *iface)
 	return status;
 }
 
+int net_if_connect(struct net_if *iface)
+{
+	int status = 0;
+
+	NET_DBG("iface %p connect", iface);
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+	if (!net_if_flag_is_set(iface, NET_IF_UP)) {
+		status = -ESHUTDOWN;
+		goto out;
+	}
+
+	/* Question to reviewers:
+	 * Would it be cleaner to attach the underlying API implementation to the
```

Comment:
I really like the idea of a separated net_l2_connectivity API, not only for the reasons you list, but also since it creates a true structural difference between ifaces that have a concept of connectivity, and those that don't. I'll give implementing that a shot next week and see how we like it, unless @jukkar or @wbober are not so convinced

As for the macrobatics, if we make it an optional param, would it be better to define a few alternate _CONNECTIVITY macros that take the extra argument, or really go for a `__VA_ARGS__` approach?

Reply:
I think `struct net_l2_connectivity` is an elegant solution.